### ISSUE TITLE
Tests fail

### ISSUE THREAD

**Issue by piotr1212** [hydrogen18/stalecucumber#8]
writer test fails on writing the nested struct. Seems like the struct is skipped and fieldnummers get mixed up, see float 23.37 became int 23. 

go version go1.6.3 linux/amd64

```
go test          
--- FAIL: TestRoundTripStruct (0.00s)
	pickle_writer_test.go:373: 
		---EXPECTED:(struct_export_test.Struct1)
		{{meow 23.37 0} 23 42.12 0 0}
		---GOT:(struct_export_test.Struct1)
		{{ 0 0} 23 42.12 0 0}
FAIL
exit status 1
FAIL	github.com/hydrogen18/stalecucumber	0.005s
```

**Comment by hydrogen18** [comment]
This fails on 1.7 for me as well, I'll investigate it more soon.

**Comment by hydrogen18** [comment]
The nested struct appears to not be getting handled properly

**Comment by hydrogen18** [comment]
It looks like somehow the code is just encoding the outer struct only and skipping the nested one entirely.

I took the pickle output from the test and loaded it using python

```
pickle.loads(base64.b16decode('80027d2858090000004c61776e6d6f7765724a1700000058090000004475706c69636174654740450f5c28f5c28f752e'.upper()))
{u'Lawnmower': 23, u'Duplicate': 42.12}
```

**Comment by hydrogen18** [comment]
In previous versions of go, the embedded structs fields basically just showed up as fields of the parent.

Now, the embedded struct is its own field.

**Comment by hydrogen18** [comment]
This should be fixed  in cd9ec28f35da4da94a4bc35ac94d7427bc5b2c78 let me know if you have any more problems.

**Comment by piotr1212** [comment]
Great thanks!
